src/test/java/core/basesyntax/service/impl/ValidatorCsvImplTest.java

```diff
@@ -0,0 +1,200 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.Validator;
+import core.basesyntax.service.activitiy.ActivityHandler;
+import core.basesyntax.service.activitiy.ActivityType;
+import core.basesyntax.service.activitiy.AddingHandler;
+import core.basesyntax.service.activitiy.RemovingHandler;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class ValidatorCsvImplTest {
+    private static Validator validator;
+    private static Map<String, ActivityHandler> activityHandlerMap;
+    private static List<String> fileData;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void beforeClass() {
+        activityHandlerMap = new HashMap<>();
+        activityHandlerMap.put(ActivityType.BALANCE.getLetter(), new AddingHandler());
+        activityHandlerMap.put(ActivityType.SUPPLY.getLetter(), new AddingHandler());
+        activityHandlerMap.put(ActivityType.RETURN.getLetter(), new AddingHandler());
+        activityHandlerMap.put(ActivityType.PURCHASE.getLetter(), new RemovingHandler());
+        validator = new ValidatorCsvImpl(activityHandlerMap);
+        fileData = new ArrayList<>();
+    }
+
+    @Before
+    public void setUp() {
+        fileData.clear();
+        fileData.addAll(List.of("type,fruit,quantity",
+                "b,banana,20",
+                "b,apple,100",
+                "s,banana,100",
+                "s,apple,13",
+                "r,banana,10",
+                "r,apple,20",
+                "p,banana,5",
+                "p,apple,50"));
```

Comment:
```suggestion
        fileData = List.of("type,fruit,quantity",
                "b,banana,20",
                "b,apple,100",
                "s,banana,100",
                "s,apple,13",
                "r,banana,10",
                "r,apple,20",
                "p,banana,5",
                "p,apple,50");
```

Reply:
Then I will get UnsupportedOperationException in future